cmd/copyEnumeratorHelper.go

```diff
@@ -84,46 +84,73 @@ var infoCopyFromBucketDirectoryListOfFiles = "trying to copy the source as bucke
 var infoCopyFromDirectoryListOfFiles = "trying to copy the source as directory/list of files"
 var infoCopyFromAccount = "trying to copy the source account"
```

Comment:
Sorry to mention it, but there seem to be some cases that are not getting caught by this code.  The ones I've spotted are:

1. We also enumerate blob storage in copyS2SMigrationBlobEnumerator.go. But that code does not call this routine, so the changes in this PR currently are not helping with the enumeration in copyS2SMigrationBlobEnumerator.go.
2. Sometimes, depending on how AzCopy is called, the code here does not execute.  E.g. when I sign in with the wrong tenant, this code in this file here executes if I run `azcopy.exe cp "https://<account>.dfs.core.windows.net/test1/foo.txt" d:\testdata\foo --recursive` but it does _not_ execute if I run `azcopy.exe cp "https://<account>.dfs.core.windows.net/test1" d:\testdata\foo --recursive`

sooo…. (and this is just a half-baked idea) how about making the "handleSingleFileValidation*" methods so they _just_ focus on returning "stop".  But move the new handling for outputting an informative message out of this file, and put it at line 896 of copy.go.  There, I think it will catch every error. (You may or may not have to take nesting of error objects into account.  If you're lucky, the simple cast to azBlob/File/bfs.StorageError will take care of grabbing the right nested error for you).  If that works, it will be guaranteed to catch everything.

Reply:
This seems to be totally resolved in the latest commit.